xmppserver/src/main/webapp/muc-room-affiliations.jsp

```diff
@@ -84,53 +84,53 @@
             errors.put("userJID","userJID");
         }
 
-        if (errors.size() == 0) {
+        if (errors.isEmpty()) {
             try {
-                ArrayList<String> memberJIDs = new ArrayList<String>();
+                List<JID> memberJIDs = new ArrayList<>();
                 if (userJID != null) {
+                    final JID addMe;
                     // Escape username
                     if (userJID.indexOf('@') == -1) {
                         String username = JID.escapeNode(userJID);
                         String domain = webManager.getXMPPServer().getServerInfo().getXMPPDomain();
+                        addMe = new JID(username, domain, null);
                         userJID = username + '@' + domain;
                     }
                     else {
                         String username = JID.escapeNode(userJID.substring(0, userJID.indexOf('@')));
-                        String rest = userJID.substring(userJID.indexOf('@'), userJID.length());
+                        String rest = userJID.substring(userJID.indexOf('@') + 1);
+                        addMe = new JID(username, rest.trim(), null);
                         userJID = username + rest.trim();
```

Comment:
Delete the userJID variable now? It doesn't appear to be used

Reply:
I initially had that, but it is used somewhere way down the page. I actually reinstated it to prevent an issue there.